src/main/java/org/example/movieSpring/controller/MovieController.java

```diff
@@ -0,0 +1,51 @@
+package org.example.movieSpring.controller;
+
+import org.example.movieSpring.model.Movie;
+import org.example.movieSpring.service.MovieService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping(path = "api/v1")
+public class MovieController {
+
+    private final MovieService movieService;
+
+    @Autowired
+    public MovieController(MovieService movieService) {
+        this.movieService = movieService;
+    }
+
+    @GetMapping("/getMovies")
+    public List<Movie> getMovies() {
+        return movieService.getMovies();
+    }
+
+    @GetMapping("/getMovie/{id}")
+    public Optional<Movie> getMovie(@PathVariable Long id) {
+        return movieService.getMovieById(id);
+    }
```

Comment:
rest controllers should never return Optionals

Reply:
Service layer returns Optional but in controller layer I check if its null and return Movie if not, do I understnad correct?

Reply:
![image](https://github.com/fstetlerKnightec/MovieProject/assets/152986163/143492e0-525c-42d6-85fc-ffa246e500a2)

Like this?

Reply:
![image](https://github.com/fstetlerKnightec/MovieProject/assets/152986163/d99a171d-cbf7-4903-8636-c01fb06ee919)

Like this?

Reply:
Yupp that is the way to go

Reply:
done